### PR TITLE
GH1492: Add ChocolateyNew Alias

### LINKED ISSUE #1492: Add ChocolateyNew Alias
Add a new alias for creating new Chocolatey packages running the `choco new` command.

### PR DESCRIPTION
**Pull request by pascalberger**
Add new alias for running `choco new`.

Fixes #1492

### CODE REVIEW

File: src/Cake.Common/Tools/Chocolatey/New/ChocolateyNewSettings.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+        /// <summary>
+        /// Gets or sets the arguments for running the installer silently.
+        /// </summary>
+        public string SilentArgs { get; set; }
+
+        /// <summary>
+        /// Gets the list of additional property values which should be passed to the template.
+        /// </summary>
+        public Dictionary<string, string> AdditionalPropertyValues
```
Comment (gep13) [inline src/Cake.Common/Tools/Chocolatey/New/ChocolateyNewSettings.cs]: Where can the user specify what these values are?  There only seems to be a get.

File: src/Cake.Common/Tools/Chocolatey/New/ChocolateyNewSettings.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    /// Contains settings used by <see cref="ChocolateyScaffolder"/>.
+    /// </summary>
+    public sealed class ChocolateyNewSettings : ChocolateySettings
+    {
+        private readonly Dictionary<string, string> _additionalPropertyValues = new Dictionary<string, string>();
+
+        /// <summary>
+        /// Gets or sets the path where the package will be created.
+        /// </summary>
+        public string OutputDirectory { get; set; }
```
Comment (gep13) [inline src/Cake.Common/Tools/Chocolatey/New/ChocolateyNewSettings.cs]: This should perhaps be a DirectoryPath, rather than a string?

**Comment by gep13** [comment]
@pascalberger in principal, I have no objection to this getting included, but I am curious as to the use case in the context of a build which is orchestrated by Cake.  What are you doing that requires the use of the new command?

File: src/Cake.Common/Tools/Chocolatey/New/ChocolateyNewSettings.cs (hunk 2 above)
Comment (pascalberger) [inline src/Cake.Common/Tools/Chocolatey/New/ChocolateyNewSettings.cs]: Changed this to a `DirectoryPath`

**Comment by gep13** [comment]
@pascalberger also, was there an issue created for this PR?

File: src/Cake.Common/Tools/Chocolatey/New/ChocolateyNewSettings.cs (hunk 1 above)
Comment (pascalberger) [inline src/Cake.Common/Tools/Chocolatey/New/ChocolateyNewSettings.cs]: I added an example how to pass additional property values: https://github.com/cake-build/cake/pull/1491/files#diff-8a3170c83272d3dbc465d927298e1c14R1023

While the Dictionary instance is readonly the user still can add entries to it from a Cake script.

**Comment by pascalberger** [comment]
@gep13 Use case is with Cake used as a task runner for scaffolding new repositories / projects.

No, I haven't created an issue yet. Should I add one?

**Comment by gep13** [comment]
>@pascalberger said...
Use case is with Cake used as a task runner for scaffolding new repositories / projects.

Ah, I see.  Interesting.  Makes sense to me!

>@pascalberger said...
Should I add one?

If you could yes.  We use GitReleaseManager to generate the release notes for the milestone, and we prefer that this linked through to an issue.

**Comment by pascalberger** [comment]
@gep13 Here you go: https://github.com/cake-build/cake/issues/1492

File: src/Cake.Common/Tools/Chocolatey/New/ChocolateyNewSettings.cs (hunk 1 above)
Comment (gep13) [inline src/Cake.Common/Tools/Chocolatey/New/ChocolateyNewSettings.cs]: Gotcha! 👍

**Comment by pascalberger** [comment]
@gep13 Added some test cases

**Comment by pascalberger** [comment]
@gep13 Let me know if I need to rebase

**Comment by gep13** [comment]
>@pascalberger said...
Let me know if I need to rebase

Ha ha, great minds...

I was just in the process of doing just that.  Will merge once the CI builds pass. 👍

**Pull request merged**